### PR TITLE
weston: 5.0.0 -> 6.0.0

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

https://lists.freedesktop.org/archives/wayland-devel/2019-March/040356.html
https://lists.freedesktop.org/archives/wayland-devel/2019-March/040334.html

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Tested a handful of the executables, not sure how to use all of them :3.

### CODE REVIEW

**Review by primeos — COMMENTED** [review]
Added a few notes but I didn't test it yet.

**Comment by Ekleog** [comment]
@dtzWill Is this still WIP?

**Comment by dtzWill** [comment]
> @dtzWill Is this still WIP?

I suspect no, but don't remember O:).

File: pkgs/applications/window-managers/weston/default.nix [hunk 1]
```diff
@@ -1,44 +1,50 @@
...
+    "-Dbackend-rdp=${boolToString (freerdp != null)}"
+    "-Dscreenshare=true"
+    "-Dweston-launch=true"
+    "-Dremoting=false" # TODO
+    "-Dsimple-dmabuf-drm=auto"
+    "-Dtest-junit-xml=false"
+    #"--enable-clients"
+    #"--disable-setuid-install" # prevent install target to chown root weston-launch, which fails
+    "-Dbackend-drm-screencast-vaapi=${boolToString (vaapi != null)}"
+  ] ++ stdenv.lib.optionals (xwayland != null) [
```
Comment (jtojnar) [inline pkgs/applications/window-managers/weston/default.nix]:
```suggestion
  ] ++ optionals (xwayland != null) [
```

**Comment by primeos** [comment]
@dtzWill also 2/4 of my review notes are still not marked as resolved, could you have a quick look at them?

I could help testing this when it's finished.

**Comment by primeos** [comment]
Just recalled that there was this PR. I'll have a quick look.

File: pkgs/applications/window-managers/weston/default.nix [hunk 2]
```diff
@@ -1,44 +1,43 @@
...
-  ] ++ stdenv.lib.optional (freerdp != null) "--enable-rdp-compositor"
-    ++ stdenv.lib.optional (vaapi != null) "--enable-vaapi-recorder"
-    ++ stdenv.lib.optionals (xwayland != null) [
-        "--enable-xwayland"
-        "--with-xserver-path=${xwayland.out}/bin/Xwayland"
-      ];
+  mesonFlags= [
+    "-Dbackend-rdp=${boolToString (freerdp != null)}"
+    "-Dremoting=false" # TODO
+    "-Dsimple-dmabuf-drm=" # Disables all drivers
```
Comment (primeos) [inline pkgs/applications/window-managers/weston/default.nix]: Note: The default is `[ 'intel', 'freedreno', 'etnaviv' ]` and I didn't check whether disabling this makes sense or not...

**Comment by primeos** [comment]
Gave it a quick test and at least `weston` and `weston-terminal` still work. But I cannot test all build configurations, hardware setups, and executables.

Testing is welcome :) I'm not sure how solid this actually is, but I'm in favor of merging it after some comments/reviews.

@dtzWill Can you rebase to resolve the merge conflict? I didn't try but AFAIK I wouldn't have the permissions anyway.

File: pkgs/applications/window-managers/weston/default.nix [hunk 3]
```diff
@@ -1,44 +1,43 @@
...
+    "-Dtest-junit-xml=false"
+    #"--enable-clients"
```
Comment (primeos) [inline pkgs/applications/window-managers/weston/default.nix]: This was introduced here https://gitlab.freedesktop.org/wayland/weston/commit/60c15080ab24ae2f4347fb04e39f147c51656688 and the clients are defined here: https://gitlab.freedesktop.org/wayland/weston/blob/6.0.0/Makefile.am#L565

File: pkgs/applications/window-managers/weston/default.nix [hunk 4]
```diff
@@ -1,44 +1,43 @@
...
+    #"--disable-setuid-install" # prevent install target to chown root weston-launch, which fails
```
Comment (primeos) [inline pkgs/applications/window-managers/weston/default.nix]: Should be fine to remove since the build doesn't fail.

**Review by primeos — APPROVED** [review]
File diff from diffoscope:

```
--- /nix/store/dfbdv03q88v28xh6vv8jz3r85aibdx10-weston-5.0.0
+++ /nix/store/2z706w77w9b8c28w335psal9wybci1r2-weston-6.0.0
├── file list
│ @@ -1,90 +1,108 @@
│  bin
│  bin/wcap-decode
│  bin/weston
│ +bin/weston-calibrator
│ +bin/weston-clickdot
│ +bin/weston-cliptest
│ +bin/weston-confine
│ +bin/weston-debug
│ +bin/weston-dnd
│ +bin/weston-editor
│ +bin/weston-eventdemo
│ +bin/weston-flower
│ +bin/weston-fullscreen
│ +bin/weston-image
│  bin/weston-info
│  bin/weston-launch
│ +bin/weston-multi-resource
│ +bin/weston-presentation-shm
│ +bin/weston-resizor
│ +bin/weston-scaler
│ +bin/weston-screenshooter
│ +bin/weston-simple-damage
│ +bin/weston-simple-dmabuf-drm
│ +bin/weston-simple-dmabuf-egl
│ +bin/weston-simple-dmabuf-v4l
│ +bin/weston-simple-egl
│ +bin/weston-simple-shm
│ +bin/weston-simple-touch
│ +bin/weston-smoke
│ +bin/weston-stacking
│ +bin/weston-subsurfaces
│  bin/weston-terminal
│ +bin/weston-touch-calibrator
│ +bin/weston-transformed
│  include
│ -include/libweston-5
│ -include/libweston-5/compositor-drm.h
│ -include/libweston-5/compositor-fbdev.h
│ -include/libweston-5/compositor-headless.h
│ -include/libweston-5/compositor-rdp.h
│ -include/libweston-5/compositor-wayland.h
│ -include/libweston-5/compositor-x11.h
│ -include/libweston-5/compositor.h
│ -include/libweston-5/config-parser.h
│ -include/libweston-5/libweston-desktop.h
│ -include/libweston-5/matrix.h
│ -include/libweston-5/plugin-registry.h
│ -include/libweston-5/timeline-object.h
│ -include/libweston-5/version.h
│ -include/libweston-5/windowed-output-api.h
│ -include/libweston-5/xwayland-api.h
│ -include/libweston-5/zalloc.h
│ +include/libweston-6
│ +include/libweston-6/compositor-drm.h
│ +include/libweston-6/compositor-fbdev.h
│ +include/libweston-6/compositor-headless.h
│ +include/libweston-6/compositor-rdp.h
│ +include/libweston-6/compositor-wayland.h
│ +include/libweston-6/compositor-x11.h
│ +include/libweston-6/compositor.h
│ +include/libweston-6/config-parser.h
│ +include/libweston-6/libweston-desktop.h
│ +include/libweston-6/matrix.h
│ +include/libweston-6/plugin-registry.h
│ +include/libweston-6/timeline-object.h
│ +include/libweston-6/version.h
│ +include/libweston-6/windowed-output-api.h
│ +include/libweston-6/xwayland-api.h
│ +include/libweston-6/zalloc.h
│  include/weston
│  include/weston/ivi-layout-export.h
│  include/weston/weston.h
│  lib
│ -lib/libweston-5
│ -lib/libweston-5.la
│ -lib/libweston-5.so
│ -lib/libweston-5.so.0
│ -lib/libweston-5.so.0.0.0
│ -lib/libweston-5/drm-backend.la
│ -lib/libweston-5/drm-backend.so
│ -lib/libweston-5/fbdev-backend.la
│ -lib/libweston-5/fbdev-backend.so
│ -lib/libweston-5/gl-renderer.la
│ -lib/libweston-5/gl-renderer.so
│ -lib/libweston-5/headless-backend.la
│ -lib/libweston-5/headless-backend.so
│ -lib/libweston-5/rdp-backend.la
│ -lib/libweston-5/rdp-backend.so
│ -lib/libweston-5/wayland-backend.la
│ -lib/libweston-5/wayland-backend.so
│ -lib/libweston-5/x11-backend.la
│ -lib/libweston-5/x11-backend.so
│ -lib/libweston-5/xwayland.la
│ -lib/libweston-5/xwayland.so
│ -lib/libweston-desktop-5.la
│ -lib/libweston-desktop-5.so
│ -lib/libweston-desktop-5.so.0
│ -lib/libweston-desktop-5.so.0.0.0
│ +lib/libweston-6
│ +lib/libweston-6.so
│ +lib/libweston-6.so.0
│ +lib/libweston-6.so.0.0.0
│ +lib/libweston-6/drm-backend.so
│ +lib/libweston-6/fbdev-backend.so
│ +lib/libweston-6/gl-renderer.so
│ +lib/libweston-6/headless-backend.so
│ +lib/libweston-6/rdp-backend.so
│ +lib/libweston-6/wayland-backend.so
│ +lib/libweston-6/x11-backend.so
│ +lib/libweston-6/xwayland.so
│ +lib/libweston-desktop-6.so
│ +lib/libweston-desktop-6.so.0
│ +lib/libweston-desktop-6.so.0.0.0
│  lib/pkgconfig
│ -lib/pkgconfig/libweston-5.pc
│ -lib/pkgconfig/libweston-desktop-5.pc
│ +lib/pkgconfig/libweston-6.pc
│ +lib/pkgconfig/libweston-desktop-6.pc
│  lib/pkgconfig/weston.pc
│  lib/weston
│ -lib/weston/desktop-shell.la
│ +lib/weston/cms-colord.so
│ +lib/weston/cms-static.so
│  lib/weston/desktop-shell.so
│ -lib/weston/fullscreen-shell.la
│  lib/weston/fullscreen-shell.so
│ -lib/weston/hmi-controller.la
│  lib/weston/hmi-controller.so
│ -lib/weston/ivi-shell.la
│  lib/weston/ivi-shell.so
│ -lib/weston/screen-share.la
│  lib/weston/screen-share.so
│ +lib/weston/systemd-notify.so
│  libexec
│  libexec/weston-desktop-shell
│  libexec/weston-ivi-shell-user-interface
│  libexec/weston-keyboard
│ -libexec/weston-screenshooter
│  libexec/weston-simple-im
│  share
│  share/man
│  share/man/man1
│ +share/man/man1/weston-debug.1.gz
│  share/man/man1/weston.1.gz
│  share/man/man5
│  share/man/man5/weston.ini.5.gz
│  share/man/man7
│  share/man/man7/weston-drm.7.gz
│  share/man/man7/weston-rdp.7.gz
│ +share/pkgconfig
│ +share/pkgconfig/libweston-6-protocols.pc
│  share/wayland-sessions
│  share/wayland-sessions/weston.desktop
│  share/weston
│  share/weston/background.png
│  share/weston/border.png
│  share/weston/fullscreen.png
│  share/weston/home.png
│ @@ -95,14 +113,16 @@
│  share/weston/icon_ivi_simple-egl.png
│  share/weston/icon_ivi_simple-shm.png
│  share/weston/icon_ivi_smoke.png
│  share/weston/icon_terminal.png
│  share/weston/icon_window.png
│  share/weston/panel.png
│  share/weston/pattern.png
│ +share/weston/protocols
│ +share/weston/protocols/weston-debug.xml
│  share/weston/random.png
│  share/weston/sidebyside.png
│  share/weston/sign_close.png
│  share/weston/sign_maximize.png
│  share/weston/sign_minimize.png
│  share/weston/terminal.png
│  share/weston/tiling.png
```

Seems like we now export additional binaries to `/bin` but that's strange since `--enable-clients` was provided before...

Overall it looks ok and should hopefully be fine.

**Comment by primeos** [comment]
For the log:

File differences with:
```
postPatch = ''
       substituteInPlace meson.build --replace "version: '6.0.0'" "version: '5.0.0'"
       substituteInPlace meson.build --replace "libweston_major = 6" "libweston_major = 5"
'';
```

# westonLite

## file list

    @@ -1,87 +1,78 @@
     bin
     bin/wcap-decode
     bin/weston
    +bin/weston-calibrator
    +bin/weston-debug
     bin/weston-info
     bin/weston-launch
    +bin/weston-screenshooter
     bin/weston-terminal
    +bin/weston-touch-calibrator
     include
     include/libweston-5
     include/libweston-5/compositor-drm.h
     include/libweston-5/compositor-fbdev.h
     include/libweston-5/compositor-headless.h
    -include/libweston-5/compositor-rdp.h
     include/libweston-5/compositor-wayland.h
     include/libweston-5/compositor-x11.h
     include/libweston-5/compositor.h
     include/libweston-5/config-parser.h
     include/libweston-5/libweston-desktop.h
     include/libweston-5/matrix.h
     include/libweston-5/plugin-registry.h
     include/libweston-5/timeline-object.h
     include/libweston-5/version.h
     include/libweston-5/windowed-output-api.h
    -include/libweston-5/xwayland-api.h
     include/libweston-5/zalloc.h
     include/weston
     include/weston/ivi-layout-export.h
     include/weston/weston.h
     lib
     lib/libweston-5
    -lib/libweston-5.la
     lib/libweston-5.so
     lib/libweston-5.so.0
     lib/libweston-5.so.0.0.0
    -lib/libweston-5/drm-backend.la
     lib/libweston-5/drm-backend.so
    -lib/libweston-5/fbdev-backend.la
     lib/libweston-5/fbdev-backend.so
    -lib/libweston-5/gl-renderer.la
     lib/libweston-5/gl-renderer.so
    -lib/libweston-5/headless-backend.la
     lib/libweston-5/headless-backend.so
    -lib/libweston-5/wayland-backend.la
     lib/libweston-5/wayland-backend.so
    -lib/libweston-5/x11-backend.la
     lib/libweston-5/x11-backend.so
    -lib/libweston-5/xwayland.la
    -lib/libweston-5/xwayland.so
    -lib/libweston-desktop-5.la
     lib/libweston-desktop-5.so
     lib/libweston-desktop-5.so.0
     lib/libweston-desktop-5.so.0.0.0
     lib/pkgconfig
     lib/pkgconfig/libweston-5.pc
     lib/pkgconfig/libweston-desktop-5.pc
     lib/pkgconfig/weston.pc
     lib/weston
    -lib/weston/desktop-shell.la
    +lib/weston/cms-colord.so
    +lib/weston/cms-static.so
     lib/weston/desktop-shell.so
    -lib/weston/fullscreen-shell.la
     lib/weston/fullscreen-shell.so
    -lib/weston/hmi-controller.la
     lib/weston/hmi-controller.so
    -lib/weston/ivi-shell.la
     lib/weston/ivi-shell.so
    -lib/weston/screen-share.la
     lib/weston/screen-share.so
    +lib/weston/systemd-notify.so
     libexec
     libexec/weston-desktop-shell
     libexec/weston-ivi-shell-user-interface
     libexec/weston-keyboard
    -libexec/weston-screenshooter
    -libexec/weston-simple-im
     share
     share/man
     share/man/man1
    +share/man/man1/weston-debug.1.gz
     share/man/man1/weston.1.gz
     share/man/man5
     share/man/man5/weston.ini.5.gz
     share/man/man7
     share/man/man7/weston-drm.7.gz
    +share/pkgconfig
    +share/pkgconfig/libweston-5-protocols.pc
     share/wayland-sessions
     share/wayland-sessions/weston.desktop
     share/weston
     share/weston/background.png
     share/weston/border.png
     share/weston/fullscreen.png
     share/weston/home.png
    @@ -92,14 +83,16 @@
     share/weston/icon_ivi_simple-egl.png
     share/weston/icon_ivi_simple-shm.png
     share/weston/icon_ivi_smoke.png
     share/weston/icon_terminal.png
     share/weston/icon_window.png
     share/weston/panel.png
     share/weston/pattern.png
    +share/weston/protocols
    +share/weston/protocols/weston-debug.xml
     share/weston/random.png
     share/weston/sidebyside.png
     share/weston/sign_close.png
     share/weston/sign_maximize.png
     share/weston/sign_minimize.png
     share/weston/terminal.png
     share/weston/tiling.png

# weston

## file list

    @@ -1,13 +1,17 @@
     bin
     bin/wcap-decode
     bin/weston
    +bin/weston-calibrator
    +bin/weston-debug
     bin/weston-info
     bin/weston-launch
    +bin/weston-screenshooter
     bin/weston-terminal
    +bin/weston-touch-calibrator
     include
     include/libweston-5
     include/libweston-5/compositor-drm.h
     include/libweston-5/compositor-fbdev.h
     include/libweston-5/compositor-headless.h
     include/libweston-5/compositor-rdp.h
     include/libweston-5/compositor-wayland.h
    @@ -23,68 +27,57 @@
     include/libweston-5/xwayland-api.h
     include/libweston-5/zalloc.h
     include/weston
     include/weston/ivi-layout-export.h
     include/weston/weston.h
     lib
     lib/libweston-5
    -lib/libweston-5.la
     lib/libweston-5.so
     lib/libweston-5.so.0
     lib/libweston-5.so.0.0.0
    -lib/libweston-5/drm-backend.la
     lib/libweston-5/drm-backend.so
    -lib/libweston-5/fbdev-backend.la
     lib/libweston-5/fbdev-backend.so
    -lib/libweston-5/gl-renderer.la
     lib/libweston-5/gl-renderer.so
    -lib/libweston-5/headless-backend.la
     lib/libweston-5/headless-backend.so
    -lib/libweston-5/rdp-backend.la
     lib/libweston-5/rdp-backend.so
    -lib/libweston-5/wayland-backend.la
     lib/libweston-5/wayland-backend.so
    -lib/libweston-5/x11-backend.la
     lib/libweston-5/x11-backend.so
    -lib/libweston-5/xwayland.la
     lib/libweston-5/xwayland.so
    -lib/libweston-desktop-5.la
     lib/libweston-desktop-5.so
     lib/libweston-desktop-5.so.0
     lib/libweston-desktop-5.so.0.0.0
     lib/pkgconfig
     lib/pkgconfig/libweston-5.pc
     lib/pkgconfig/libweston-desktop-5.pc
     lib/pkgconfig/weston.pc
     lib/weston
    -lib/weston/desktop-shell.la
    +lib/weston/cms-colord.so
    +lib/weston/cms-static.so
     lib/weston/desktop-shell.so
    -lib/weston/fullscreen-shell.la
     lib/weston/fullscreen-shell.so
    -lib/weston/hmi-controller.la
     lib/weston/hmi-controller.so
    -lib/weston/ivi-shell.la
     lib/weston/ivi-shell.so
    -lib/weston/screen-share.la
     lib/weston/screen-share.so
    +lib/weston/systemd-notify.so
     libexec
     libexec/weston-desktop-shell
     libexec/weston-ivi-shell-user-interface
     libexec/weston-keyboard
    -libexec/weston-screenshooter
    -libexec/weston-simple-im
     share
     share/man
     share/man/man1
    +share/man/man1/weston-debug.1.gz
     share/man/man1/weston.1.gz
     share/man/man5
     share/man/man5/weston.ini.5.gz
     share/man/man7
     share/man/man7/weston-drm.7.gz
     share/man/man7/weston-rdp.7.gz
    +share/pkgconfig
    +share/pkgconfig/libweston-5-protocols.pc
     share/wayland-sessions
     share/wayland-sessions/weston.desktop
     share/weston
     share/weston/background.png
     share/weston/border.png
     share/weston/fullscreen.png
     share/weston/home.png
    @@ -95,14 +88,16 @@
     share/weston/icon_ivi_simple-egl.png
     share/weston/icon_ivi_simple-shm.png
     share/weston/icon_ivi_smoke.png
     share/weston/icon_terminal.png
     share/weston/icon_window.png
     share/weston/panel.png
     share/weston/pattern.png
    +share/weston/protocols
    +share/weston/protocols/weston-debug.xml
     share/weston/random.png
     share/weston/sidebyside.png
     share/weston/sign_close.png
     share/weston/sign_maximize.png
     share/weston/sign_minimize.png
     share/weston/terminal.png
     share/weston/tiling.png

**Comment by primeos** [comment]
It still builds/runs, `nix-review` was successful, and looking at the file changes it should be fine. I added myself as maintainer in case anything breaks.

**Comment by dtzWill** [comment]
On Fri, 31 May 2019 14:00:02 -0700, Michael Weiss <notifications@github.com> wrote:
> Merged #58522 into master.

Thank you for seeing this through, sorry for not assisting more.

\o/ this is finally merged :D.

**Pull request merged**